### ISSUE TITLE
update rubocop to a current version

### ISSUE THREAD

**Issue by freibuis** [puppetlabs/puppet-module-gems#115]
```yaml
dev:
  shared:
    gem: rubocop
    version: '~> 0.49.0'
```
`0.49.0` has been out for almost 2 years now. version `0.74.0` has been out for over a month

the old version plays havoc with new gems like solargraph which require at least 0.54 to work correctly

**Comment by logicminds** [comment]
0.50 breaks requires ruby >= 2.3 which breaks support

**Comment by benjarrell** [comment]
Are there still supported versions with ruby < 2.3? It looks to like there is not, based on https://puppet.com/docs/pe/latest/component_versions_in_recent_pe_releases.html

**Comment by freibuis** [comment]
current puppet uses ruby version 2.5.7
rubocop will not work with this version without disabling ruby version flags.

```bash
puppet --version
6.14.0
```

```bash
/opt/puppetlabs/puppet/bin/ruby -v
ruby 2.5.7p206 (2019-10-01 revision 67816) [x86_64-linux]
```

```bash
bundle exec rubocop --version
0.49.1
```
running rubocop the pdk way (gitlab ci rubocop)
```bash
bundle exec rake rubocop
Known versions: 1.9, 2.0, 2.1, 2.2, 2.3, 2.4
```
can we get it upgraded now?

**Comment by DavidS** [comment]
Updating rubocop will require a coordinated upgrade path to update the profiles defined in pdk-templates in all modules with minimal disturbance.

**Comment by shepting** [comment]
I came to create this very same issue but saw that it was already open. 

I had a what I thought was a very safe assumption that `gem 'puppet-module-posix-dev-r2.6'` would be able to use a modern version of Rubocop:

![Screen Shot 2020-07-09 at 11 04 58 AM](https://user-images.githubusercontent.com/48065/87074670-0b989d00-c1d4-11ea-8b1c-3b8126ffd65f.png)

**Comment by logicminds** [comment]
I think the only way to do this because rubocop has many changes for each version is to keep a separate template based on ruby version. 

So the puppet-module-gems version would be able to pick the correct rubycop.yml template per ruby.  Additionally, we can use the [mry](puppet-module-gems) tool to easily upgrade rubocop changes.

**Comment by DavidS** [comment]
This has been addressed in #156 and will be released later today as 1.0.0, with an accompanying change to pdk-[templates](https://github.com/puppetlabs/pdk-templates/pull/371)
